sky/clouds/aws.py

```diff
@@ -495,7 +516,9 @@ def get_current_user_identity(cls) -> Optional[List[str]]:
             # userid changed for a cluster).
             # 2. In the case where the multiple users belong to an organization,
             # those users will have different account id, so fallback works.
-            user_ids = [user_info['UserId'], user_info['Account']]
+            user_ids = [
+                user_info['UserId'], user_info['Account'], user_info['Arn']
```

Comment:
Is this change safe? Do people who upgrade past this branch need to run `status -r` explicitly?